specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerEffectiveConfiguration.json

```diff
@@ -0,0 +1,180 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "NetworkManagerEffectiveConfiguration",
+    "description": "The Microsoft Azure Network Manager effective configuration API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
+    "version": "2021-02-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{virtualNetworkName}/networkManagerEffectiveConfigurations": {
+      "parameters": [
+        {
+          "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+        },
+        {
+          "$ref": "#/parameters/ResourceGroupNameParameter"
+        },
+        {
+          "$ref": "#/parameters/VirtualNetworkNameParameter"
+        },
+        {
+          "$ref": "./network.json#/parameters/ApiVersionParameter"
+        },
+        {
+          "$ref": "#/parameters/ListPageSizeParameter"
+        },
+        {
+          "$ref": "#/parameters/ListSkipTokenParameter"
+        }
+      ],
+      "get": {
+        "tags": [
+          "NetworkManagerEffectiveConfiguration"
+        ],
+        "operationId": "EffectiveConfigurations_List",
+        "description": "List all configurations in a virtual network.",
+        "responses": {
+          "200": {
+            "description": "OK - Returns information about the configurations.",
+            "schema": {
+              "$ref": "#/definitions/NetworkManagerEffectiveConfigurationListResult"
+            }
+          },
+          "default": {
+            "description": "Resource Provider error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List effective configuration": {
+            "$ref": "./examples/NetworkManagerEffectiveConfigurationsList.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "EffectiveConfiguration": {
+      "properties": {
+        "configType": {
```

Comment:
As per our earlier conversation, I understand currently you don't have a way to generate a id, name for items in the GET collection. But this is something you need to think about and expose during your GA. ARM does a RPC strong validation and fire alerts for the APIs that do not follow the contact. Check link https://armwiki.azurewebsites.net/api_contracts/RPCstrongvalidation.html#get-resource-collection

Reply:
We will consider about it, but keep it as it is for now